src/v1/app.layout.ts

```diff
@@ -53,27 +53,48 @@ export class LayoutManager<T> {
         });
     }
 
-    addEditor(name: string, lang: string, isReadOnly: boolean = false, callback: ((editor: AceAjax.Editor) => void) = null) {
+    addEditor(name: string, lang: string, isReadOnly: boolean = false, callback: ((editor: monaco.editor.IStandaloneCodeEditor) => void) = null) {
         this.addComponent(name, container => {
-            var editor = ace.edit(container.getElement().get(0));
-            editor.setTheme("ace/theme/monokai");
-            editor.getSession().setMode(`ace/mode/${lang}`);
-            if (lang === "yaml")
-                editor.setOption("tabSize", 2);
-            editor.$blockScrolling = Infinity; // TODO: remove this line after they fix ACE not to throw warning to the console
-            editor.setReadOnly(isReadOnly);
-            if (callback)
-                callback(editor);
+            let editor;
+            if (name === "ksyEditor") {
+                const uri = monaco.Uri.parse("inmemory://model/main");
+                editor = monaco.editor.create(container.getElement().get(0), {
+                    language: lang,
+                    theme: "vs-dark",
+                    showFoldingControls: "always",
+                    model: monaco.editor.createModel("", lang, uri)
+                });
+            } else {
+                editor = monaco.editor.create(container.getElement().get(0), {
+                    language: lang,
+                    theme: "vs-dark",
+                });
+            }
+
+            console.log(editor.getModel().uri.toString());
+            editor.getModel().updateOptions({ tabSize: 2 });
+
+
+            // var editor = ace.edit(container.getElement().get(0));
+            // editor.setTheme("ace/theme/monokai");
+            // editor.getSession().setMode(`ace/mode/${lang}`);
+            // if (lang === "yaml")
+            //     editor.setOption("tabSize", 2);
+            // editor.$blockScrolling = Infinity; // TODO: remove this line after they fix ACE not to throw warning to the console
+            // editor.setReadOnly(isReadOnly);
+            // if (callback)
+            //     callback(editor);
```

Comment:
probably can be just deleted; we can totally rely on git to keep version history for us